### ISSUE TITLE
auto-approve for `terragrunt destroy` not working

### ISSUE THREAD

**Issue by gercograndia** [gruntwork-io/terragrunt#3119]
## Describe the bug

After upgrading to the latest version of terragrunt, when running `terragrunt destroy -auto-approve` it still keeps asking for approval.

## Steps To Reproduce

```console
❯ terragrunt --version
terragrunt version v0.58.2

❯ terragrunt destroy -auto-approve --terragrunt-no-auto-approve=false
Detected dependent modules:
xxx
WARNING: Are you sure you want to continue? (y/n)
```

## Expected behavior

If you pass the -auto-approve flag, it should destroy without asking for approval.

Using plain `terraform` is does without complaining:

```console
❯ terraform --version
Terraform v1.5.7
on darwin_arm64
+ provider registry.terraform.io/hashicorp/azuread v2.48.0
+ provider registry.terraform.io/hashicorp/azurerm v3.101.0
+ provider registry.terraform.io/hashicorp/random v3.6.1
+ provider terraformprovider.z6.web.core.windows.net/eon/eoncms v5.0.1

Your version of Terraform is out of date! The latest version
is 1.8.2. You can update by downloading from https://www.terraform.io/downloads.html

❯ terraform destroy -auto-approve -var-file terragrunt-debug.tfvars.json
...
Destroy complete! Resources: 15 destroyed.
```

## Versions

- Terragrunt version: see before
- OpenTofu/Terraform version: see before
- Environment details (Ubuntu 20.04, Windows 10, etc.): macos Sonoma

**Comment by CristianoOliveira1** [comment]
worked for me..
terragrunt destroy -auto-approve

and for all..
terragrunt run-all --terragrunt-non-interactive destroy

**Comment by gercograndia** [comment]
So you're saying the auto-approve works as intended but if there are dependent modules there will always be a request for confirmation?

So in my view the auto-approve is doing half of its job then.

**Comment by CristianoOliveira1** [comment]
yes, was also surprised with that. behaviour.  but when using destroy for the whole set of modules, ( "terragrunt run-all --terragrunt-non-interactive destroy" )  it worked.

**Comment by gercograndia** [comment]
Ok, I do believe this is somewhat a workaround and the `auto-approve` should just auto-approve.

But glad there is a workaround at least, so we can move on.

Thanks for your input!

**Comment by J02670** [comment]
Hi everybody,

I guess this happens because `auto-approve` is not a Terragrunt flag. Terragrunt passes `auto-approve` to Terraform, probably without any interpretation.

So what you can do (e. g. if you run Terragrunt/Terraform in a CI/CD environment) without human intervention possible: Use the Terragrunt CLI flag `--terragrunt-non-interactive`: https://terragrunt.gruntwork.io/docs/reference/cli-options/#terragrunt-non-interactive

From the documentation:

> When passed in, don’t show interactive user prompts. This will default the answer for all Terragrunt (not OpenTofu/Terraform) prompts to yes except for the listed cases below. This is useful if you need to run Terragrunt in an automated setting (e.g. from a script).

This resolved the issue for me.

So in conclusion, your call to Terragrunt should be like this:
`terragrunt --terragrunt-non-interactive destroy -auto-approve`
